pkg/api/write_test.go

```diff
@@ -248,6 +249,47 @@ func TestWrite(t *testing.T) {
 	}
 }
 
+// BenchmarkProtoUnmarshall is primarily focused on benchmarking
+// memory usage and allocations when unmarshalling WriteRequests
+func BenchmarkProtoUnmarshall(b *testing.B) {
+	seriesGen, err := testsupport.NewSeriesGenerator(100, 100, 5)
+	if err != nil {
+		b.Error(err)
+	}
+	batches := seriesGen.GetTimeseriesInBatch(1000)
+	mock := &mockInserter{}
+	metrics = &Metrics{LastRequestUnixNano: 0}
+	dataParser := parser.NewParser()
+	numSamplesReceived := &mockMetric{}
+	writeHandler := Write(mock, dataParser, mockUpdaterForIngest(&mockMetric{}, nil, numSamplesReceived, nil))
+
+	b.ResetTimer()
+	b.ReportAllocs()
+	for i := 0; i < b.N; i++ {
+		for _, batch := range batches {
+			b.StopTimer()
+			req, err := testsupport.GetHTTPWriteRequest("", &prompb.WriteRequest{
```

Comment:
We should create and pre-allocate these write requests before we call b.ResetTimer() because othewise these allocations are included in the benchmark results, making them harded to interpret

Reply:
`StopTimer` turns of allocations/mem counter so we should be good I believe. I also tested without `StopTimer` and noticed more allocations.

Reply:
Also I had that done initially but since here we iterate `b.N` we might re-use HTTP request which we should not.